### ISSUE TITLE
There should be cop to detect method call using class name in the same class.

### ISSUE THREAD

**Issue by akshaymohite** [bbatsov/rubocop#1369]
There should be cop to detect method call with class in name from the class method of the same class. E.g.
```Ruby
def TestClass
  def self.some_method_a
    # some code
    some_method_c
  end

  def self.some_method_b
    # some code
    TestClass.some_method_c
  end

  def self.some_method_c
    # some code
  end
end
```
Cop should detect that `TestClass.some_method_c` calling method `some_method_c` should be directly instead of `ClassName.method_name`.
As if the method `some_method_c` would have been private, this call `TestClass.some_method_c` will give error on the method call.

**Comment by alexdowad** [comment]
This could change behavior in subclasses of `TestClass`.

**Comment by jonas054** [comment]
@alexdowad You're absolutely right! I missed that in my earlier comment.
```ruby
class TestClass
  def self.some_method_a
    # some code
    some_method_c
  end

  def self.some_method_b
    # some code
    TestClass.some_method_c
  end

  def self.some_method_c
    puts 'TestClass.some_method_c'
  end
end

class Sub < TestClass
  def Sub.some_method_c
    puts 'Sub.some_method_c'
  end
end

TestClass.some_method_a # => TestClass.some_method_c
TestClass.some_method_b # => TestClass.some_method_c
Sub.some_method_a       # => Sub.some_method_c
Sub.some_method_b       # => TestClass.some_method_c
```
Notice how the calls on the `Sub` class give different results when you call `some_method_a` and `some_method_b`.

So I don't think we can support this. Closing.
